regress/regress_index.sql

```diff
@@ -34,14 +34,14 @@ set enable_bitmapscan = off;
 set enable_seqscan = on;
 
 SELECT 'scan_idx', qnodes('select * from test where the_geom && ST_MakePoint(0,0)');
- select num,ST_astext(the_geom) from test where the_geom && 'BOX3D(125 125,135 135)'::box3d order by num;
+ select num,ST_astext(the_geom) from test where the_geom::box3d && 'BOX3D(125 125,135 135)'::box3d order by num;
```

Comment:
Indeed, this is to get around the spgist storage issue; the plan was to generalize the index once pg11 is out.

Reply:
@alesuiss Pg11 is out: https://apt.postgresql.org/pub/repos/apt/pool/11/p/postgresql-11/ or https://wiki.postgresql.org/wiki/Apt/FAQ#I_want_to_try_the_beta_version_of_the_next_PostgreSQL_release

Reply:
I meant released (we're currently basing our work on the last released version).

Should we work with pg11 instead for this patch?

Reply:
If that is not too much trouble for you. :)
Postgres 11 beta1 was released just a couple of days ago: https://www.postgresql.org/about/news/1855/ and is aimed at developers.

Basically, a released non-beta Postgres 11 means a released non-beta PostGIS 2.5, and feature freeze is going to come soon.

I'll see if I can get it running on Travis to make testing easier.

Reply:
We are working on porting this Pull Request to the beta version of PostgreSQL 11. We will let you now soon.

Reply:
We have tested the Pull Request with postgresql-11beta1 and postgis-2.5.0beta1dev. Could you please elaborate what we should do next ?

Reply:
In that case we would need to define new operator names since, e.g., && and &&& are used for geometries working, respectively, on box2df and GIDX bounding boxes. How about using something like

	OPERATOR        1        <<< ,
	OPERATOR        2        &<< ,
	OPERATOR        3        &/&  ,
	OPERATOR        4        &>> ,
	OPERATOR        5        >>> ,
	OPERATOR        6        ~== ,
	OPERATOR        7        @>> ,
	OPERATOR        8        <<@	 ,
	OPERATOR        9        &<<| ,
	OPERATOR        10       <<<| ,
	OPERATOR        11       |>>> ,
	OPERATOR        12       |&>> ,
	OPERATOR        29       &<</ ,
	OPERATOR        30       <<</ ,
	OPERATOR        31       />>> ,
	OPERATOR        32       /&>> ,

Reply:
Any interest in just limiting to the frequently used ops for how? overlaps, contains, within?

Reply:
Indeed we can start with only a few operators. 

However, please notice that all this pull request originates from working with trajectories, i.e., temporal points and thus we have developed an extension based on PostGIS in which we defined temporal points, as well as temporal floats (e.g., for expressing the distance between moving cars) and temporal booleans (e.g., for determining whether a moving car is located within a province). In that case we used the z dimension for encoding timestamps and thus the operators such as <</ and />>correspond to before and after i time and these are heavily used in that scenario.

Reply:
Interesting. Did you know that there is some trajectory work in PostGIS already, encoding the timestamps on the M?

Reply:
Yes of course, but didn't scale up well, in particular because time is ordered by definition while M dimension is not and therefore we cannot leverage on that for having a much faster implementation

Reply:
BTW, please notice that the bug I reported
https://trac.osgeo.org/postgis/ticket/4088#ticket
has not been corrected in the beta version of postgis 2.5.0
I am also aiming to develop spgist 2D indexes on geometries using boxd2f bounding boxes. However, the fact that the over* (e.g. overleft) operators have erroneously defined a commutative operator  prevents me to have correct results.

Reply:
We should have the generalized index ready soon. Should we use a new branch & new pull request for this?